src/Castle.Core/Compatibility/NetCoreComponentModelExtensions.cs

```diff
@@ -12,11 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#if FEATURE_NETCORE_CONVERTER_API
+#if FEATURE_NETCORE_COMPONENTMODEL_API
 
-namespace System
+namespace System.ComponentModel
 {
-    public delegate TOutput Converter<TInput, TOutput>(TInput input);
+	public interface IDataErrorInfo
```

Comment:
I'm not that comfortable publicly exposing this interface in Castle.Core especially under the System.ComponentModel namespace. We don't own it and nothing else on .NET Core is going to be able to use ours so I see no point in having it.

Outside of our unit tests there are only 3 usages of the interface, but possibly many more of the members. I'm not sure exactly what to do as I don't use DA. Do you know how much work it would be to conditionally compile out the interface usage but leave the implementation so consumers can call the properties directly?

My preference with .NET Core is to exclude any functionality we can't implementation rather than put more stop gap solutions into place, we can always expose more functionality later but much harder to remove hacky things like our own custom IDataErrorInfo interface.

Thoughts?